### PR TITLE
#54 - Added ManifestRef.string() method

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #54 
Added method to represent manifest reference as a string. This is required to generate `Location` header when manifest is added to the registry.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#102](https://github.com/artipie/docker-adapter/issues/102) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/9d776a61-b0e2-4c86-8167-347e1d00dc20); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/docker/ref/ManifestRefTest.java [hunk 1]
```diff
@@ -95,4 +95,21 @@ void resolvesTagLink() {
...
     }
+
+    @Test
+    void stringFromDigestRef() {
+        MatcherAssert.assertThat(
+            new ManifestRef.FromDigest(new Digest.Sha256("0123")).string(),
+            Matchers.equalTo("sha256:0123")
+        );
+    }
+
```
Comment (Vatavuk) [inline src/test/java/com/artipie/docker/ref/ManifestRefTest.java:106]: @olegmoz add one test for `FromString`

Comment (olegmoz) [reply src/test/java/com/artipie/docker/ref/ManifestRefTest.java:106]: @Vatavuk thanks, added a test for this class too

**Comment by 0crat** [comment]
Job was finished in 44 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@1141192d-e349-458f-8a77-e0fe31931131, thread: PQ-CT2E6TK9B
com.zerocracy.farm.ruled.RdRules[130] java.lang.IllegalStateException: http://datum.zerocracy.com/latest/rules/pm/no-lost-boosts.xsl says: The boost for job "gh:artipie/docker-adapter#102" exists in boosts.xml, but the job is absent in the WBS among 57 jobs.
```

[0.53.14](https://github.com/zerocracy/farm): CID: [0cf03fbb-84b3-4f39-a068-b9c44e2e390f](https://www.0crat.com/footprint/CT2E6TK9B/0cf03fbb-84b3-4f39-a068-b9c44e2e390f), Type: "Remove job from WBS"

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**